tests/api2/test_virt_vm.py

```diff
@@ -70,6 +70,39 @@ def test_iso_import_as_volume(virt_pool):
         assert vol['content_type'] == 'ISO'
 
 
+@pytest.mark.parametrize('vol_name, should_work', [
+    (
+        '123invalid', False
+    ),
+    (
+        '-invlaid-name', False
+    ),
+    (
+        'valid-name', True
+    ),
+    (
+        'invalid-name-', False
+    ),
+    (
+        'volume-name-should-not-have-characters-more-than-sixty-three-characters--',
+        False
+    ),
+    (
+        'alpine-3.18-default.iso', True
+    ),
+    (
+        'alpine-3.18-default-x86_64.iso', True
+    )
+])
+def test_volume_name_validation(virt_pool, vol_name, should_work):
```

Comment:
I'm all for adding tests but we really have to consider the length of time our tests are taking. I would rather test the explicit limitations that you're adding with this PR. This means you should test just 2x failure conditions.
1. test it fails when volume name is > 63 chars
2. test a volume name that doesn't adhere to the regex